flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionAndRetrievalITCase.java

```diff
@@ -52,17 +60,28 @@ public void testLeaderElectionAndRetrieval() throws Exception {
         KubernetesLeaderElectionDriver leaderElectionDriver = null;
         KubernetesLeaderRetrievalDriver leaderRetrievalDriver = null;
 
+        final FlinkKubeClient flinkKubeClient = kubernetesResource.getFlinkKubeClient();
+        final Configuration configuration = kubernetesResource.getConfiguration();
+
+        final String clusterId = configuration.getString(KubernetesConfigOptions.CLUSTER_ID);
+        final KubernetesConfigMapSharedWatcher configMapSharedWatcher =
+                flinkKubeClient.createConfigMapSharedWatcher(
+                        KubernetesUtils.getConfigMapLabels(
+                                clusterId, LABEL_CONFIGMAP_TYPE_HIGH_AVAILABILITY));
+        configMapSharedWatcher.run();
+        final ExecutorService watchExecutorService = Executors.newCachedThreadPool();
```

Comment:
I guess we either need to shut the pool down after tests are finished or have a daemon thread factory, otherwise this might block jvm from being shut down.